### PR TITLE
Nz/3558 Write Cypress tests for basic APD functions #3558

### LINKED ISSUE #3558: [Feature] Write Cypress tests for basic APD functions
**Description and related issues -**
We needs cypress tests to test the basics of creating and navigating around an APD.

### Test cases needed...

- [x] create an APD
- [x] verify that side nav buttons take you to the correct section
- [x] verify that navigation buttons at the bottom of each section of the APD take you to the correct section
- [x] delete the APD and verify that it was removed from the dashboard

### PR DESCRIPTION
**Pull request by Sun-Mountain**
Resolves #3558 

**Description-**
We need cypress tests to test the basics of creating and navigating around an APD.

**This pull request changes...**

- 01-apd-basics.js
  - Created tests for:
    - creating APD
    - verify that side nav buttons work
    - verify that anchor links work
    - verify Continue & Before buttons at bottom of each section work
    - delete APD and verify it was removed

**This pull request also touches…**

- Updated `node:14` to `node:14-bullseye` for most recent python version
- Added `react` to `package.json` dependencies
- Added new commands to `integrationTests/cypress/support/commands.js` for anchor link navigation.

**This pull request was tested in the follow ways…**

- list of frontend cases
- list of backend cases

**Steps to manually verify this change...**

1. steps to view and verify change

### This pull request is ready to review when...

- [x] Automated tests are updated (and all tests are passing)
- [ ] The change has been documented
- [ ] Associated OpenAPI documentation has been updated
- [ ] Changelog is updated as appropriate
- [ ] The experience passes a basic manual accessibility audit (keyboard nav, screenreader, text scaling) OR an exemption is documented

### This pull request can be merged when…

- [x] Code has been reviewed by someone other than the original author
- [ ] QA has verified the accessibility and functionality related to the change
- [ ] Design has approved the experience
- [ ] Product has approved the experience

### CODE REVIEW

File: integrationTests/cypress/support/commands.js [hunk 1]
```diff
@@ -160,6 +160,23 @@ Cypress.Commands.add('waitForSave', () => {
     });
 });
```
Comment (tbolt) [inline integrationTests/cypress/support/commands.js:162]: For these new commands could we have an array/object which contains the nav items then map over that to create the commands? Might make it easier to add/remove nav items if the app changes.

File: integrationTests/cypress/integration/apd/01-apd-basics.js [hunk 2]
```diff
@@ -2,4 +2,217 @@
...
+      cy.get('.ds-h2')
+        .should('contain', index);
+
+      if(index != reverse_page_titles[reverse_page_titles.length - 1]) {
+        cy.get('#previous-button').click();
+      }
+    });
+  });
+
+  it('confirms side nav buttons redirect to correct sections', () => {
```
Comment (tbolt) [inline integrationTests/cypress/integration/apd/01-apd-basics.js:56]: Not critical but we could probably have an object of the nav items and loop over them. Similar to the command comment I had below. Maybe there's a way to do it with cypress fixtures or pages, maybe @thetif knows.

**Review by tbolt — APPROVED** [review]
Looks good! Made an optional suggestion

File: integrationTests/cypress/integration/apd/01-apd-basics.js (hunk 2 above)
Comment (Sun-Mountain) [reply integrationTests/cypress/integration/apd/01-apd-basics.js:56]: I took both notes and applied them!

**Pull request merged**